internal/pkg/deploy/cloudformation/workload.go

```diff
@@ -63,5 +63,9 @@ func (cf CloudFormation) handleStackError(stackName string, err error) error {
 
 // DeleteWorkload removes the CloudFormation stack of a deployed workload.
 func (cf CloudFormation) DeleteWorkload(in deploy.DeleteWorkloadInput) error {
-	return cf.cfnClient.DeleteAndWait(fmt.Sprintf("%s-%s-%s", in.AppName, in.EnvName, in.Name))
+	stackName := fmt.Sprintf("%s-%s-%s", in.AppName, in.EnvName, in.Name)
```

Comment:
what about `stack.NameForService`? If it gets shortened by `stack.NameForService` when it was created/updated, we should use the same stack name here.